tests/test.javascript.js

```diff
@@ -0,0 +1,91 @@
+import { SEVERITY_ERROR, SEVERITY_WARNING } from 'const';
+import JavaScriptScanner from 'javascript';
+import { singleLineString } from 'utils';
+
+
+describe('JS Code Checker', function() {
+
+  it('should not allow mozIndexedDB', (done) => {
+    var code = 'var myDatabase = indexeddb || mozIndexedDB;';
+    var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+    jsScanner.scan()
+      .then((validationMessages) => {
+        assert.equal(validationMessages.length, 1);
+        assert.equal(validationMessages[0].id, 'mozIndexedDB');
+        assert.equal(validationMessages[0].severity, SEVERITY_ERROR);
+        done();
+      });
+  });
+
+  // Warn here, as it's possible to have a property named "mozIndexedDB" that
+  // isn't referencing window.mozIndexedDB. The add-on reviewer can manually
+  // check this. It should NOT output an error.
+  it('should warn when mozIndexedDB is used as a property', (done) => {
+    var code = 'var myDatabase = window.mozIndexedDB;';
+    var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+    jsScanner.scan()
+      .then((validationMessages) => {
+        assert.equal(validationMessages.length, 1);
+        assert.equal(validationMessages[0].id, 'mozIndexedDB_prop');
+        assert.equal(validationMessages[0].severity, SEVERITY_WARNING);
+        done();
+      });
+  });
+  it('should warn when mozIndexedDB is a square-bracket key', (done) => {
+    var code = 'var myDatabase = window["mozIndexedDB"];';
+    var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+    jsScanner.scan()
+      .then((validationMessages) => {
+        assert.equal(validationMessages.length, 1);
+        assert.equal(validationMessages[0].id, 'mozIndexedDB_prop');
+        assert.equal(validationMessages[0].severity, SEVERITY_WARNING);
+        done();
+      });
+  });
+  it('should warn when mozIndexedDB is used as a literal', (done) => {
+    var code = 'var foo = "mozIndexedDB"; var myDatabase = window[foo];';
+    var jsScanner = new JavaScriptScanner(code, 'badcode.js');
+
+    jsScanner.scan()
+      .then((validationMessages) => {
+        assert.equal(validationMessages.length, 1);
+        assert.equal(validationMessages[0].id, 'mozIndexedDB_prop');
+        assert.equal(validationMessages[0].severity, SEVERITY_WARNING);
+        done();
+      });
+  });
+  // Not sure how to test for this one yet; it's pretty messy.
+  it.skip('should warn when mozIndexedDB is assembled into literal', (done) => {
```

Comment:
This is pretty important, if eslint can't provide a good way to test for this this validation is easily subverted.